src/platform/android/android.c

```diff
@@ -0,0 +1,216 @@
+        result = (float) (*handler.env)->CallFloatMethod(handler.env, handler.activity, handler.method);
+    }
+    destroy_java_function_handler(&handler);
+    return result;
+}
+
+int android_get_file_descriptor(const char *filename, const char *mode)
+{
+    int result = 0;
+    java_function_handler handler;
+    if (!request_java_static_function_handler("com/github/bvschaik/julius/FileManager", "openFileDescriptor", "(Lcom/github/bvschaik/julius/JuliusMainActivity;Ljava/lang/String;Ljava/lang/String;)I", &handler)) {
+        destroy_java_function_handler(&handler);
+        return 0;
+    }
+    jstring jfilename = (*handler.env)->NewStringUTF(handler.env, filename);
+    jstring jmode = (*handler.env)->NewStringUTF(handler.env, mode);
+    result = (int) (*handler.env)->CallStaticIntMethod(handler.env, handler.class, handler.method, handler.activity, jfilename, jmode);
+    (*handler.env)->DeleteLocalRef(handler.env, jfilename);
+    (*handler.env)->DeleteLocalRef(handler.env, jmode);
+    destroy_java_function_handler(&handler);
+
+    return result;
+}
+
+int android_set_base_path(const char *path)
+{
+    int result = 0;
+    java_function_handler handler;
+    if (!request_java_static_function_handler("com/github/bvschaik/julius/FileManager", "setBaseUri", "(Ljava/lang/String;)I", &handler)) {
+        destroy_java_function_handler(&handler);
+        return 0;
+    }
+    jstring jpath = (*handler.env)->NewStringUTF(handler.env, path);
+    result = (int) (*handler.env)->CallStaticIntMethod(handler.env, handler.class, handler.method, jpath);
+    (*handler.env)->DeleteLocalRef(handler.env, jpath);
+    destroy_java_function_handler(&handler);
+
+    return result;
+}
+
+int android_get_directory_contents(const char *dir, int type, const char *extension, int (*callback)(const char *))
+{
+    java_function_handler handler;
+    if (!request_java_static_function_handler("com/github/bvschaik/julius/FileManager", "getDirectoryFileList", "(Lcom/github/bvschaik/julius/JuliusMainActivity;Ljava/lang/String;ILjava/lang/String;)[Ljava/lang/String;", &handler)) {
+        destroy_java_function_handler(&handler);
+        return LIST_ERROR;
+    }
+    jstring jdir = (*handler.env)->NewStringUTF(handler.env, dir);
+    jstring jextension = (*handler.env)->NewStringUTF(handler.env, extension);
+    jobjectArray result = (jobjectArray) (*handler.env)->CallStaticObjectMethod(handler.env, handler.class, handler.method, handler.activity, jdir, type, jextension);
+    (*handler.env)->DeleteLocalRef(handler.env, jdir);
+    (*handler.env)->DeleteLocalRef(handler.env, jextension);
+    int match = LIST_NO_MATCH;
+    int len = (*handler.env)->GetArrayLength(handler.env, result);
+    for (int i = 0; i < len; ++i) {
+        jstring jfilename = (jstring) (*handler.env)->GetObjectArrayElement(handler.env, result, i);
+        const char *filename = (*handler.env)->GetStringUTFChars(handler.env, jfilename, NULL);
```

Comment:
All `GetStringUTFChars` are memleaking as they must be released by calling `ReleaseStringUTFChars`, so I must fix this before merge.